src/js/common/schemaform/SelectiveArrayField.jsx

```diff
@@ -0,0 +1,203 @@
+
+  render() {
+    const {
+      uiSchema,
+      errorSchema,
+      idSchema,
+      formData,
+      disabled,
+      readonly,
+      registry,
+      formContext,
+      onBlur,
+      schema
+    } = this.props;
+    const definitions = registry.definitions;
+    const { TitleField, SchemaField } = registry.fields;
+
+    const title = uiSchema['ui:title'] || schema.title;
+    const hideTitle = !!_.get(['ui:options', 'hideTitle'], uiSchema);
+    const description = uiSchema['ui:description'];
+    const textDescription = typeof description === 'string' ? description : null;
+    const DescriptionField = typeof description === 'function'
+      ? uiSchema['ui:description']
+      : null;
+    const hasTitleOrDescription = (!!title && !hideTitle) || !!description;
+    const selectedFields = uiSchema['ui:selectedFields'] || Object.keys(schema.properties);
+
+    // if we have form data, use that, otherwise use an array with a single default object
+    const items = (formData && formData.length)
+      ? formData
+      : [getDefaultFormState(schema, undefined, registry.definitions)];
+
+    // uiSchema's ui:order complains loudly if there are extraineous properties
+    //  or missing properties. We take the ui:order and slap on the selectedFields
+    //  to make sure we're not missing any. _.intersection() will take the
+    //  unique values in the order in which they occur in ui:order.
+    if (uiSchema.items && uiSchema.items['ui:order']) {
+      uiSchema.items['ui:order'] = _.intersection(
+        // Make sure ALL the selectedFields are in there
+        _.concat(uiSchema.items['ui:order'], uiSchema['ui:selectedFields']),
+        uiSchema['ui:selectedFields']
+      );
+    }
+
+    let containerClassNames = classNames({
+      'schemaform-field-container': true,
+      'schemaform-block': hasTitleOrDescription
+    });
+
+
+    return (
+      <div className={containerClassNames}>
+        {hasTitleOrDescription && <div className="schemaform-block-header">
+          {title && !hideTitle
+              ? <TitleField
+                  id={`${idSchema.$id}__title`}
+                  title={title}
+                  formContext={formContext}/> : null}
+          {textDescription && <p>{textDescription}</p>}
+          {DescriptionField && <DescriptionField options={uiSchema['ui:options']}/>}
+          {!textDescription && !DescriptionField && description}
+        </div>}
+        <div>
+          {items.map((item, index) => {
+            const itemSchema = this.getItemSchema(index, selectedFields);
+            const itemIdPrefix = `${idSchema.$id}_${index}`;
+            const itemIdSchema = toIdSchema(itemSchema, itemIdPrefix, definitions);
+
+            // Use ui:setItemTitle function to set a title for each item
+            // Note: this overrides the ui:title
+            // Also, we have to create a new object or we'll have trouble with the reassignment
+            let itemUiSchema = _.assign({}, uiSchema.items);
```

Comment:
You could do this in an `updateSchema` function and set `schema.title` instead. I'd like to keep the modifications to uiSchema to a minimum (or none, ideally), since it's the sort of blueprint for all the changes that happen to the regular schema. All of this becomes even harder to follow when the blueprint is changing, too.